src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -77,12 +77,22 @@ void AdaptDispatch::_WriteToBuffer(const std::wstring_view string)
     const auto wrapAtEOL = _api.GetAutoWrapMode();
     const auto attributes = textBuffer.GetCurrentAttributes();
 
+    const auto viewport = _api.GetViewport();
+    const auto [topMargin, bottomMargin] = _GetVerticalMargins(viewport, true);
+    const auto [leftMargin, rightMargin] = _GetHorizontalMargins(textBuffer.GetSize().Width());
+
+    auto lineWidth = textBuffer.GetLineWidth(cursorPosition.y);
+    if (cursorPosition.x <= rightMargin && cursorPosition.y >= topMargin && cursorPosition.y <= bottomMargin)
```

Comment:
What we're calculating here is whether the cursor should wrap at the edge of the screen or at the right margin, and the left margin is of no relevance to that. You wouldn't want the text to bypass the right margin, just because the initial cursor position was outside the left margin.